src/metabase/native_query_analyzer/replacement.clj

```diff
@@ -85,13 +104,21 @@
                   (update-keys clean-table)
                   (update-vals clean-table))}))
 
+(defn- param-values
+  [query]
+  (if (qp.store/initialized?)
+    (params.values/query->params-map (:native query))
+    (qp.setup/with-qp-setup [q query]
+      (params.values/query->params-map (:native q)))))
```

Comment:
I think we could DRY this up a bit:

```suggestion
  (let [query (if (qp.store/initialized?)
                query
                (qp.setup/with-qp-setup [q query] q))
    (params.values/query->params-map (:native query)))
```

Perhaps better with inlining, just looked a bit ugly on the tiny screen I'm using.

Reply:
`with-qp-setup` (irritatingly) binds dynamic variables for its `& body`, so that doesn't work :(

Reply:
Should be fine if you inline the binding though? Your call